_posts/2019-10-15-dependency-management.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: blog-detail
+post-type: blog
+by: Alexandre Archambault, Sébastien Doeraene
+title: "Better management of transitive dependencies and conflicts"
+---
+
+Early in 2019, the Scala Center advisory board accepted [a proposal from Spotify](https://github.com/scalacenter/advisoryboard/blob/master/proposals/020-sbt-transitive-dependencies-conflicts.md), whose goal is to improve the handling of conflicts in dependency management in sbt.
+
+The proposal revolves around 3 axes:
+
+* improving the documentation about how versions are reconciled in sbt, and what to do about conflicts,
+* have a usable strict conflict manager in sbt, and
+* develop tools relying on static analysis to detect runtime conflicts between libraries that would not have been detected otherwise.
+
+At the time we started this project, Ivy was still used by default to resolve libraries in sbt, but efforts were in progress to use coursier instead, so we focused on the latter as a vehicle for improvements.
+Since then, coursier support in sbt [became official](https://github.com/sbt/sbt/pull/4614), and ships by default in sbt 1.3.0.
+Besides having a simpler model than the one used before in sbt or in Ivy, coursier has the advantage of being usable from the command line, and can benefit other users of the coursier API.
+
+For the third point about static analysis, we found [the missinglink tool](https://github.com/spotify/missinglink) for Maven, originating from Spotify themselves.
+This tool can analyze the classpath of a Maven project, and detect ahead of time (at "link" time) whether binary incompatibilities can be encountered at run-time.
+The only issue was that it was not usable from sbt, rendering it useless for a large subset of Scala users.
+We worked on [an sbt plugin for missinglink](https://github.com/scalacenter/sbt-missinglink) to address this hole.
+
+## Version ordering and reconciliation
+
+While most people have a simple intuition about library resolution (a newer version of a library get picked over an older version), the details of how versions are compared and reconciled are quite complicated.
+We contributed [a dedicated documentation page](https://get-coursier.io/docs/other-version-handling) on the coursier website to help users understand how resolution works in the not-so-trivial cases.
+In order to improve the experience of upgrading from sbt 1.2.x and Ivy-based resolution, we [tweaked resolution in coursier 2.0.0-RC3-4](https://github.com/coursier/coursier/pull/1348) to be more in line with Ivy, the former sbt behavior, and the [semantic versioning specification](https://semver.org).
```

Comment:
I'd say it's "we", with the help of Eugene Yokota. (The commits in https://github.com/coursier/coursier/pull/1348 originate from earlier changes he proposed.)